graphql/src/test/java/com/linecorp/armeria/server/graphql/GraphqlServiceTest.java

```diff
@@ -249,4 +266,60 @@ void shouldReturnBadRequestForInvalidQuery() {
         assertThat(response.status()).isEqualTo(HttpStatus.BAD_REQUEST);
         assertThat(response.contentUtf8()).contains("Validation error of type FieldUndefined");
     }
+
+    @Test
+    void shouldAllowUndefinedParametersQuery() {
+        final HttpRequest request = HttpRequest.builder().post("/graphql")
+                                         .content(MediaType.GRAPHQL_JSON,
+                                                  "{\"operationName\":null," +
+                                                  "\"variables\":{}," +
+                                                  "\"query\":\"query ($value: ComplexTypeInput) " +
+                                                  "{ optionalInput(value: $value) } \"}"
+                                         )
+                                         .build();
+        final AggregatedHttpResponse response = WebClient.of(server.httpUri())
+                                                   .execute(request)
+                                                   .aggregate().join();
+        assertThat(response.status()).isEqualTo(HttpStatus.OK);
+        assertThatJson(response.contentUtf8()).node("data.optionalInput").isEqualTo(UNDEFINED);
+    }
+
+    @Test
+    void shouldAllowNonnullParametersQuery() {
+        final String query =
+                "{\n" +
+                "  \"operationName\": null,\n" +
+                "  \"variables\": {\n" +
+                "    \"value\": {\n" +
+                "      \"innerValue\": \"foobar\"\n" +
+                "    }\n" +
+                "  },\n" +
+                "  \"query\": \"query ($value: ComplexTypeInput) { optionalInput(value: $value) }\"\n" +
+                "}\n";
+        final HttpRequest request = HttpRequest.builder().post("/graphql")
+                                               .content(MediaType.GRAPHQL_JSON, query)
+                                               .build();
+        final AggregatedHttpResponse response = WebClient.of(server.httpUri())
+                                                         .execute(request)
+                                                         .aggregate().join();
```

Comment:
How about using a brand new `BlockingWebClient`?
```suggestion
        final AggregatedHttpResponse response = BlockingWebClient.of(server.httpUri()).execute(request);
```

Reply:
Oh, I just copied without pride from the other test cases. I'll change it!